### PR TITLE
docs: improve JSON Schema validation

### LINKED ISSUE #4076: [Feature] completion for `nodeLinker` variants (.yarnrc.yml schema)
- [x ] I'd be willing to implement this feature ([contributing guide](https://yarnpkg.com/advanced/contributing))
- [ ] This feature is important to have in this repository; a contrib plugin wouldn't do

**Describe the user story**

According to the docs, [`nodeLinker`](https://yarnpkg.com/configuration/yarnrc#nodeLinker) can be one of: `pnp`, `pnpm` and `node-modules`. Would be great to have these values in the completion list produced by JSON Schema. The completion of possible values works great if the possible values are listed as `enum`.

**Describe the solution you'd like**

Add `pnp`, `pnpm` and `node-modules` as `enum`.

### PR DESCRIPTION
**Pull request by segrey**
**What's the problem this PR addresses?**

Fixes #4076

### CODE REVIEW

File: packages/gatsby/static/configuration/yarnrc.json [hunk 1]
```diff
@@ -414,7 +414,7 @@
       "description": "Defines what linker should be used for installing Node packages (useful to enable the node-modules plugin), one of: `pnp`, `pnpm` and `node-modules`.",
       "type": "string",
       "default": "pnp",
-      "examples": ["pnp", "pnpm", "node-modules"]
+      "enum": ["pnp", "pnpm", "node-modules"]
```
Comment (merceyz) [inline packages/gatsby/static/configuration/yarnrc.json:417]: It isn't an enum of just these three values, plugins can add more options linkers

Comment (arcanis) [reply packages/gatsby/static/configuration/yarnrc.json:417]: For the purpose of the JSON schema (which only enhances the IDE experience) it's probably fine to define them as an enum. Users can still set them to other values if they need, it'll just be displayed with yellow squiggles.

Comment (segrey) [reply packages/gatsby/static/configuration/yarnrc.json]: Didn't know `nodeLinker` can be other than "pnp", "pnpm", "node-modules" (documentation says "one of: `pnp`, `pnpm` and `node-modules`").
Fixed! Now it should accept any string value and autocomplete will show known values: "pnp", "pnpm", "node-modules".

Comment (segrey) [reply packages/gatsby/static/configuration/yarnrc.json]: The solution is taken from https://github.com/json-schema-org/json-schema-spec/issues/665

Comment (merceyz) [reply packages/gatsby/static/configuration/yarnrc.json]: Perfect, could you update `pnpMode` while you're at it?

> which only enhances the IDE experience

Note that VSCode already shows these three values as auto-complete options

Comment (segrey) [reply packages/gatsby/static/configuration/yarnrc.json]: @merceyz As I can see, `pnpMode` is already `"enum": ["strict", "loose"]` and other values are disallowed, IIUC. 

>Note that VSCode already shows these three values as auto-complete options

Maybe you're right and `examples` should be auto-completed, but it feels a bit wrong to complete an arbitrary example in general case. In this particular case, auto-completing `examples` is correct.

Comment (merceyz) [reply packages/gatsby/static/configuration/yarnrc.json]: > As I can see, pnpMode is already "enum": ["strict", "loose"] and other values are disallowed, IIUC.

Plugins can provide their own value here as well so it should follow the same config as `nodeLinker`

**Review by merceyz — APPROVED** [review]

**Pull request merged**